src/huggingface_hub/inference/_mcp/agent.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+import asyncio
+from typing import AsyncGenerator, Dict, List, Optional, Sequence, Union
+
+from huggingface_hub import ChatCompletionInputMessage, ChatCompletionStreamOutput, MCPClient
+
+from .._providers import PROVIDER_OR_POLICY_T
+from .constants import DEFAULT_SYSTEM_PROMPT, EXIT_LOOP_TOOLS, MAX_NUM_TURNS
+
+
+class Agent(MCPClient):
+    """
+    Python implementation of a Simple Agent
+    i.e. just a basic while loop on top of an Inference Client with MCP-powered tools
+    """
+
+    def __init__(
+        self,
+        *,
+        model: str,
+        servers: Sequence[Dict],
```

Comment:
```suggestion
        servers: Iterable[Dict],
```
(nit) if we want to give more flexibility than a `List`, let's set it to `Iterable` which is broader

Reply:
Otherwise `List[Dict]` is fine IMO